### ISSUE TITLE
Sender not in contact list message when opening a conversation.

### ISSUE THREAD

**Issue by bme13** [WhisperSystems/Signal-Android#7244]
I have:
- [x] searched open and closed issues for duplicates
- [x] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports

----------------------------------------

### Bug description
When I open some old 1:1 conversations from the conversation list I get the message "The sender is not in your contact list". These persons are in my contact list though and if I check my contact list, Signal is listed in the "connected via" list. Also some of them are part of Signal groups that I do use regularly. 
It can easily be resolved by choosing the "Add to contacts" option and then directly go back to Signal, without actually adding the contact so it's not a major issue however it's a bit confusing.
Attached a logfile from the moment I opened a conversation and got the "Sender not in contact list" message.
Just to specify 'old' :-) Conversations older then September show this behavior.

### Steps to reproduce
- Open old conversation

**Actual result:** Some contacts are reported not to be in the contact list when they actually are.
**Expected result:** Don't get "sender not in contact list" message for contacts that are in the contact list.

### Device info

**Device:** Samsung SM-G900F
**Android version:** 6.0.1
**Signal version:** 4.14.1

### Link to debug log

https://gist.github.com/anonymous/2f9eb52d3f69ede8cd82e1208122da28

**Comment by moxie0** [comment]
Did this just start happening with 4.14? And do you have the contacts permission enabled for Signal?

**Comment by bme13** [comment]
Sorry, I don't know if this started with 4.14.
Yes, contact permissions are enabled.

**Comment by bme13** [comment]
Just did a few tests. Here's something else I found.
I have a contact that only contained recent messages and did not show this "sender not in contact message". As a test I've removed all messages in this conversation. After that immediately the "sender not in contact message" popped up. I pressed back to leave the conversation. Since the conversation now was empty it was not listed in the main view. By pressing the compose (pencil) I searched for it and opened it. Then the "sender not in contact message" did not show up anymore. When a new message was received for that contact it now also shows this "sender not in contact message" 
Hope this helps to reproduce it.
